### PR TITLE
adding preferences flow

### PR DESCRIPTION
**Pull request by TalyaS**
[Full sequence diagram](http://swimlanes.io/#hVHLTsMwELznK1Y9ItqUipMPHFBQDghSidexMslGtZLY0a5DhSr+HbtNYQtF5LSOZ2fGM0vCGgltiQx16zZgLGR5klyT2zASTK/gDqlca+shnN+QFDzFi7I1ZcMweTa4gf6bZZL8wEeKLFewLB4eJTBlZDbOcnLvPCp4MX6dnIHuDTT4HibGktCHYYh63jVokyyPdEWmoj5av9J9z6ke/DocTKk9/sNWZAc/xS0s5vMRbmoQJMEV1Nq0DGFlIAuX8wW86gp6TbpDj8Sjk0jE3hHuTK52JmOEhJU5YH7lt1O+gC2MCaxMBR+nYhtLUFA76gK+0l5PY0tKBnkueJSYZ7NZoBVFRrP5zVEJIi25Kl7ndXP0uJpcJ0WgwjYkIn+JlqJcXOb0tKhgPq5mu78RW+Ete96vVP6Che8T)

[Sequence diagram from PoV of the merchant](http://swimlanes.io/#bZDLCoMwEEX3+Yqha7WPZRZdFEsWpSj0sZUQRxq0iSS2Uor/3ojSpuJuQu6cnJvUYIEGlUALRaVbkApiRsjO6NaigXALRzTixlUD7vxEQ+HSX4hKitLC4iqxhfpHWZBJvkfEjEKanM5+cGnRWqmVhTc8XDRrdIkqAF7LrMTXMFgUBhvoSMxmVZIDbFZrRxhhmcxdeMZg7EOh0Obu8jlveNgXpr5T4HGoN0dR5LDen/SF2P6vz1RiMP6+60zHnv5O9wE=)

### CODE REVIEW

**Comment by TalyaS** [comment]
@moriaklarna @tomerlevi444 @cdambo @danevron @guyb7 @tomeralon : Please review

File: integration.md [hunk 1]
```diff
@@ -210,6 +213,42 @@ every 1.day do
...
+
+### Acquiring a preferences session id
+Once a user has chosen to view their prefernces (by clicking a designated button, or in any other form you see fit), your backend server should acquire a preferences session identifier. This is done by calling the following method:
+
+    URL:
+      production: https://ondemand.klarna.com/web
+      playground: https://ondemand-dg.playground.klarna.com/web
+    Action: preferneces
+    Method: POST
+    Params:
```
Comment (danevron) [inline integration.md:232]: Why not doing the same setup we have in OnDemand (base64...)?

File: integration.md [hunk 2]
```diff
@@ -210,6 +213,42 @@ every 1.day do
...
+      Content: { session_id: <session_id> }
+    Error Response:
+      Code: 402
+      Content: { error : "credentials_mismatch" }
+
+### Placing a preferences form on your page
+Assuming you've already included the SDK itself on your page, place the preferences form in the page. This form will allow the user to view and change their preferences:
+
+```html
+<form action="/" method="POST" class="klarna-form" data-api-key="test_d4ca9ed6-489b-42b3-8011-dacdcee0fdb6" data-flow="preferences" data-session-id="AAA42CA8-9765-4BAB-9135-865F6C565540">
```
Comment (danevron) [inline integration.md]: Why not have the action be an empty string?

File: integration.md [hunk 3]
```diff
@@ -210,6 +213,42 @@ every 1.day do
 end
 ```
 
+## Preferences
+After is user is registered with Klarna, you might want to allow them to view and possibly change their current payment preferences.
```
Comment (moriaam) [inline integration.md]: After **a** user

File: integration.md [hunk 4]
```diff
@@ -210,6 +213,42 @@ every 1.day do
...
+Since the preferences form displays sensitive data about the user without prior authenticion, rendering the flow is a 2-step process: a preferences session is first securly created, and then the preferenecs data is displayed against the session identifier.
```
Comment (moriaam) [inline integration.md]: authenticion --> authentication
preferenecs --> preferences (this is recurring in all the document)
securly --> securely

File: integration.md [hunk 5]
```diff
@@ -220,4 +259,5 @@ data-flow | The type of on-demand form to display. The supported values are `pur
...
 data-currency | This attribute is valid only for purchase forms and denotes the purchase's currency, as an [ISO 4217 code](https://en.wikipedia.org/wiki/ISO_4217).
-data-merchant-context | This attribute is optional. We recommend sending information about the product. Klarna will use this to segment your traffic and will be able to offer ideas that may help your conversion as a merchant. A practical example is to send data about whether you're selling a "basic subscription" or "premium subscription", or if your trying "promotion banner 1" against "promotion banner 2".
\ No newline at end of file
+data-merchant-context | This attribute is optional. We recommend sending information about the product. Klarna will use this to segment your traffic and will be able to offer ideas that may help your conversion as a merchant. A practical example is to send data about whether you're selling a "basic subscription" or "premium subscription", or if your trying "promotion banner 1" against "promotion banner 2".
+data-session-id | This attribute is valid only for preferences form and denotes a preferences session identifier (used as means for displaying the preferences securly)
```
Comment (moriaam) [inline integration.md]: securly ---> securely

File: integration.md [hunk 6]
```diff
@@ -210,6 +213,42 @@ every 1.day do
...
+      user_token: the requesting user's token. Required
+    Success Response:
+      Code: 201
+      Content: { session_id: <session_id> }
+    Error Response:
+      Code: 402
+      Content: { error : "credentials_mismatch" }
+
+### Placing a preferences form on your page
+Assuming you've already included the SDK itself on your page, place the preferences form in the page. This form will allow the user to view and change their preferences:
```
Comment (danevron) [inline integration.md]: I think saying something like the following will be clearer

Place a Klarna form in your rendered page. This form will open a Klarna frame that will allow...

WDYT?

**Comment by danevron** [comment]
LGTM

**Comment by danevron** [comment]
We will want to expire the session after some time. It's worth noting this in the docs.
(Also in the implementation tickets :) ).

File: integration.md [hunk 7]
```diff
@@ -210,6 +213,42 @@ every 1.day do
...
+
+```html
+<form action="/" method="POST" class="klarna-form" data-api-key="test_d4ca9ed6-489b-42b3-8011-dacdcee0fdb6" data-flow="preferences" data-session-id="AAA42CA8-9765-4BAB-9135-865F6C565540">
+</form>
+```
+
+Mind that action is not set - since no action is required of the merchant backend once the preferences page is displayed. Also, data-flow is set to preferences and a the preferences session id is provided under the key data-session-id. [This table](#custom_form_attributes) covers data attributes' meaning.
```
Comment (moriaam) [inline integration.md]: a the

File: integration.md (hunk 1 above)
Comment (TalyaS) [inline integration.md:232]: @danevron makes perfect sense. changed.

File: integration.md (hunk 2 above)
Comment (TalyaS) [inline integration.md]: @danevron sure. changed

File: integration.md (hunk 3 above)
Comment (TalyaS) [inline integration.md]: @moriaam  done

File: integration.md (hunk 4 above)
Comment (TalyaS) [inline integration.md]: @moriaam fixed thanks

File: integration.md (hunk 5 above)
Comment (TalyaS) [inline integration.md]: @moriaam done

File: integration.md (hunk 6 above)
Comment (TalyaS) [inline integration.md]: @danevron  done

File: integration.md (hunk 7 above)
Comment (TalyaS) [inline integration.md]: done

**Comment by moriaam** [comment]
LGTM

**Pull request merged**